src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -92,25 +92,31 @@ const _calculateDistancesAlongRoute = (waypointModelList, airport) => {
     let distanceFromClosestFixToAirspaceBoundary = 0;
     let totalDistance = 0;
 
-    for (let i = 0; i < waypointModelList.length; i++) {
+    // Iteration started at index 1 to ensure two elements are available. It is
+    // already an expectation that aircraft must have two waypoints, so this
+    // should not be a problem here.
+    for (let i = 1; i < waypointModelList.length; i++) {
         const waypoint = waypointModelList[i];
-        const waypointPosition = waypoint.relativePosition;
-        let previousWaypoint = waypoint;
-        let previousPosition = waypoint.relativePosition;
+        const previousWaypoint = waypointModelList[i - 1];
 
-        if (i > 0) {
-            previousWaypoint = waypointModelList[i - 1];
-            previousPosition = previousWaypoint.relativePosition;
+        // TODO: Refactor to avoid continue statement
+        if (waypoint.isVector || previousWaypoint.isVector) {
+            continue;
```

Comment:
@n8rzz You know me, I'll do what I'm told. :laughing:

Linter said no, but if you want yes, feel free to open up the lint rc and change the rule. You'll hear no complaints from this guy, especially given the example above. :+1: